app/src/main/java/org/p2p/wallet/user/interactor/BlockChainTokensMetadataInteractor.kt

```diff
@@ -0,0 +1,43 @@
+package org.p2p.wallet.user.interactor
+
+import com.google.gson.Gson
+import timber.log.Timber
+import org.p2p.core.token.TokenData
+import org.p2p.wallet.common.feature_toggles.toggles.remote.TokenMetadataUpdateFeatureToggle
+import org.p2p.wallet.common.storage.ExternalStorageRepository
+import org.p2p.wallet.user.repository.UserLocalRepository
+import org.p2p.wallet.user.repository.UserRepository
+
+private const val TAG = "BlockChainTokensInteractor"
+
+class BlockChainTokensMetadataInteractor(
```

Comment:
BlockChain -> Blockchain

and if it's SOL - tokens, so it should be named like SolTokensMetadataInteractor because ETH is a blockchain too